### PR TITLE
Added Unity audio record but still doesn't work

### PR DESCRIPTION
**Pull request by tulingcheng86**

### CODE REVIEW

File: unity/Assets/AudioRecorder.cs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    {
+      StartRecording();
+    }
+  }
+
+  void StartRecording()
+  {
+    Debug.Log("开始录制音频");
+
+    AudioClip recordedClip = Microphone.Start(microphoneDevice, false, maxRecordTime, 44100);
```
Comment (a-wing) [inline unity/Assets/AudioRecorder.cs:33]: I think unity don't need use Microphone

File: unity/Assets/AudioRecorder.cs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    }
+
+    writer.Dispose();
+
+    return stream.ToArray();
+  }
+
+  void WriteWavHeader(Stream stream, AudioClip audioClip)
+  {
+    BinaryWriter writer = new BinaryWriter(stream);
```
Comment (a-wing) [inline unity/Assets/AudioRecorder.cs:88]: Maybe there should use audio library

File: unity/Assets/CanvasTextSender.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
+using UnityEngine;
+using WebSocketSharp;
+
+public class CanvasTextSender : MonoBehaviour
+{
+  private WebSocket ws;
+
+  private void Start()
+  {
+    ws = new WebSocket("ws://your_websocket_server_address");
```
Comment (a-wing) [inline unity/Assets/CanvasTextSender.cs]: We already have a webrtc datachannel and a websocket connection, Why need a new Webcocket?

**Review by a-wing — CHANGES_REQUESTED** [review]
Finally, The Unity need deploy to headless server

The headless server no Microphone!

* * *

We're need capture audio in Browser

After, use webrtc add a audio trick

```
Browser audio stream -> webrtc -> unity
```

Unity receive  the audio stream

Save the audio stream, and replay the audio

File: unity/Assets/AudioRecorder.cs (hunk 1 above)
Comment (tulingcheng86) [reply unity/Assets/AudioRecorder.cs:33]: can remove it

**Pull request merged**